bundles/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/se/handler/ValloxSEHandler.java

```diff
@@ -0,0 +1,477 @@
+
+    private boolean reconnect = false;
+    private byte panelNumber;
+
+    public ValloxSEHandler(Thing thing, ValloxConnector connector) {
+        super(thing);
+        this.connector = connector;
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Vallox SE handler");
+        if (watchDog != null) {
+            watchDog.cancel(true);
+            watchDog = null;
+        }
+        closeConnection();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Vallox SE handler");
+        updateStatus(ThingStatus.UNKNOWN);
+        cache.clear();
+        if (watchDog == null) {
+            watchDog = scheduler.scheduleWithFixedDelay(this::checkConnection, 0, 30, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Connect to Vallox unit and start refresher only after successful connection.
+     */
+    private void connect() {
+        if (!isConnected()) {
+            try {
+                ValloxSEConfiguration config = getConfigAs(ValloxSEConfiguration.class);
+                this.panelNumber = config.getPanelAsByte();
+                connector.addListener(this);
+                connector.connect(config);
+                if (refreshJob == null) {
+                    refreshJob = scheduler.scheduleWithFixedDelay(this::refreshChannels, 1, 5, TimeUnit.MINUTES);
+                }
+                updateStatus(ThingStatus.ONLINE);
+            } catch (IOException e) {
+                logger.debug("Connection failed ", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                reconnect = true;
+            }
+        }
+    }
+
+    /**
+     * Close the connection and stop refresh job
+     */
+    private void closeConnection() {
+        connector.removeListener(this);
+        connector.close();
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    /**
+     * Check if connection is open
+     */
+    private boolean isConnected() {
+        return connector.isConnected();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (this.thing.getStatus() == ThingStatus.ONLINE && isConnected()) {
+            ChannelDescriptor descriptor = ChannelDescriptor.get(channelUID.getId());
+            if (descriptor == ChannelDescriptor.NULL) {
+                logger.debug("Channel descriptor returned null");
+                return;
+            }
+            if (command instanceof RefreshType) {
+                handleRefreshTypeCommand(command, descriptor);
+            } else if (command instanceof DecimalType) {
+                handleDecimalCommand(((DecimalType) command).intValue(), descriptor);
+            } else if (command instanceof QuantityType) {
+                handleDecimalCommand(((QuantityType<?>) command).intValue(), descriptor);
```

Comment:
Shouldn't you make sure you convert the units correctly before trying to handle it?